public/app/plugins/datasource/cloudwatch/datasource.ts

```diff
@@ -234,64 +233,74 @@ export class CloudWatchDatasource extends DataSourceApi<CloudWatchQuery, CloudWa
         statsQuery: (param.statsGroups?.length ?? 0) > 0 ?? false,
       };
     });
+
+    const dataFrames = increasingInterval(100, 1000, 300).pipe(
```

Comment:
This would probably be more readable as named params/object instead ie `{startPeriod: 100, endPeriod: 1000, step: 300}`

Reply:
Fair enough, will change